File: services/api-nestjs/src/compliance/compliance.controller.ts
import { Controller, Get } from '@nestjs/common'
import { ApiTags } from '@nestjs/swagger'
import { Public } from '../common/public.decorator'
import { compliance } from './compliance'

// Reports the industry profile this deployment is running under, and the
// controls in effect. Switch the profile with COMPLIANCE_PROFILE; this
// endpoint proves the controls changed — no code change, no rebuild.
@ApiTags('compliance')
@Controller('compliance')
export class ComplianceController {
  @Public()
  @Get()
  current() {
    return {
      profile: compliance.profile,
      name: compliance.name,
      jurisdiction: compliance.jurisdiction,
      controls: compliance.controls,
    }
  }
}

File: services/api-spring-boot/src/main/java/com/example/controller/ComplianceController.java
package com.example.controller;

import com.example.config.Compliance;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Map;

/**
 * Reports the active industry profile and the controls in effect.
 * Switch with COMPLIANCE_PROFILE — the controls flip at boot, no rebuild.
 */
@RestController
public class ComplianceController {

    private final Compliance compliance;

    public ComplianceController(Compliance compliance) {
        this.compliance = compliance;
    }

    @GetMapping("/compliance")
    public Map<String, Object> current() {
        return Map.of(
                "profile", compliance.getProfile(),
                "controls", Map.of(
                        "auditLogging", compliance.isAuditLogging(),
                        "sessionTimeoutSeconds", compliance.getSessionTimeoutSeconds(),
                        "mfaRequired", compliance.isMfaRequired(),
                        "encryptionInTransit", compliance.isEncryptionInTransit()),
                "required", compliance.getRequired());
    }
}
